File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/FungibleResourceIndicator.kt
package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.Decimal192
import com.radixdlt.sargon.FungibleResourceIndicator
import com.radixdlt.sargon.fungibleResourceIndicatorGetAmount

val FungibleResourceIndicator.amount: Decimal192
    get() = fungibleResourceIndicatorGetAmount(indicator = this)

File: crates/sargon-uniffi/src/wrapped_radix_engine_toolkit/low_level/execution_summary/resource_indicator/fungible_resource_indicator.rs
use crate::prelude::*;
use sargon::FungibleResourceIndicator as InternalFungibleResourceIndicator;

#[derive(Clone, PartialEq, Eq, InternalConversion, uniffi::Enum)]
pub enum FungibleResourceIndicator {
    Guaranteed { decimal: Decimal },
    Predicted { predicted_decimal: PredictedDecimal },
}

#[uniffi::export]
pub fn new_fungible_resource_indicator_sample() -> FungibleResourceIndicator {
    InternalFungibleResourceIndicator::sample().into()
}

#[uniffi::export]
pub fn new_fungible_resource_indicator_sample_other(
) -> FungibleResourceIndicator {
    InternalFungibleResourceIndicator::sample_other().into()
}

#[uniffi::export]
pub fn fungible_resource_indicator_get_amount(
    indicator: &FungibleResourceIndicator,
) -> Decimal192 {
    indicator.into_internal().get_amount().into()
}
